### DISCUSSION TITLE
Tiling Shell v15.0 | Automatically tile new windows and more!

### DISCUSSION THREAD

**Discussion by domferr** [domferr/tilingshell#185] [Announcements]
- **New**: Finally, **automatic tiling**! When a window is created, it is automatically moved to the best tile according to where other windows are tiled and the current layout. The setting can be enabled/disabled from the extension's preferences (disabled by default). It follows the currently selected layout to pick the best tile for newly created windows. The best tile is the vacant tile nearest to the center of the screen. Closes #169 

https://github.com/user-attachments/assets/b54269ca-42ba-4863-97b5-10661e58da66

- **Fix**: improved user experience of keybindings. Closes #180 and #159 

<div align="center">

![](https://img.shields.io/badge/GNOME-42--47-e04196)
[![](https://img.shields.io/badge/Request_a_Feature-8A2BE2)](https://github.com/domferr/tilingshell/issues/new?assignees=&labels=&projects=&template=feature_request.md&title=)
[![](https://img.shields.io/badge/Report_a_Bug-blue)](https://github.com/domferr/tilingshell/issues/new?assignees=&labels=&projects=&template=bug_report.md&title=)
[![kofi](https://img.shields.io/badge/Donate-on_Ko--fi-purple?logo=ko-fi)](https://ko-fi.com/domferr)

</div>

## How to install

:star: This update of Tiling Shell is landing on GSE too!
<div align="center">
  <a href="https://extensions.gnome.org/extension/7065/tiling-shell/" >
      <img src="https://img.shields.io/badge/Install%20from-extensions.gnome.org-4A86CF?style=for-the-badge&logo=Gnome&logoColor=white"/>
  </a>
</div>

## Manual installation

There are two builds and you need only one of them, according to your GNOME Shell version:
- For GNOME Shell versions 45, 46 and 47: `tilingshell@ferrarodomenico.com.zip`
- For GNOME Shell versions from 42 to 44: `GNOME.42-44.tilingshell@ferrarodomenico.com.zip`

To know your GNOME Shell version, run `gnome-shell --version`.

Extract the downloaded archive. Copy its content to `~/.local/share/gnome-shell/extensions/tilingshell@ferrarodomenico.com` directory. You need to reload GNOME Shell afterwards (e.g. by logging out). Then you can enable the extension:
```
/usr/bin/gnome-extensions enable tilingshell@ferrarodomenico.com
```

<hr /><em>This discussion was created from the release <a href='https://github.com/domferr/tilingshell/releases/tag/15.0.0'>Tiling Shell v15.0 | Automatically tile new windows and more!</a>.</em>

**Comment by mpstaton** [comment]
Hey @domferr this GNOME extension was exactly what I was looking for.  It took me like 2 days to find you!  I was busy trying to get AwesomeWM working and it unfortunately is a whole other GUI.  But I finally found this after much frustration!  If I get some time (unlikely), I would love to help!!!!

Thanks a lot for this great work!  Hope you keep at it (or it gets integrated or something)....  Whatever a good trajectory is for you!

**Reply by mpstaton** [reply]
I definitely wish installing things was easier for GNOME.  (I didn't want to download Chrome again...)

**Reply by mpstaton** [reply]
Also @domferr do you have a "feature requests" kind of forum?  I don't want to open an issue because right now it's perfect for me.  But of course Keyboard Shortcuts and hotkeys or numbers for tiles would be clutch!

**Reply by domferr** [reply]
Hey thank you for the kind words! I'm so happy you are enjoying it! There are some new features coming soon, stay tuned. Feel free to open a new issue here on GitHub for any feature request or bug!
